tsdb/head.go

```diff
@@ -908,18 +928,21 @@ func (a *headAppender) log() error {
 func (a *headAppender) Commit() error {
 	defer a.head.metrics.activeAppenders.Dec()
 	defer a.head.putAppendBuffer(a.samples)
+	defer a.head.putSeriesBuffer(a.sampleSeries)
```

Comment:
Hm, what's the goal of this change? Sorry I might be  missing the context, but I think this change is totally orthogonal to extracting the code to separate packages, right? Thus, it might be the best to keep those in separate PRs.

Reply:
I was pretty against exposing `memSeries` in the last PR, and this was a way to be able to move `RefSample` to the `record` package without also having to move and expose `memSeries`. You can see some context starting with [this comment](https://github.com/prometheus/tsdb/pull/606#issuecomment-499331881), and a handful of comments afterward.

Reply:
Cool, that's a valid use case, I will looks closer to those non-move-code changes then :+1:

Reply:
LGTM